src/dune_rules/coq/coq_lib.ml

```diff
@@ -504,16 +505,30 @@ module DB = struct
     let create_from_stanza coq_db db dir stanza =
       Memo.exec memo (coq_db, db, dir, stanza)
 
-    (* XXX: Memoize? This is pretty cheap so not sure worth the cost *)
+    let directory_of_file p =
+      Path.of_string (Path.to_string p |> Filename.dirname)
+
+    (* XXX: Memoize? This is pretty cheap so not sure worth the cost,
+       still called too much I have observed, suspicious! *)
```

Comment:
This is called 3 times on a single .vo build, so that seems quite suspicious.

Reply:
Could it be that the ~boot_id is changing each time?

Reply:
Good idea, I will try.

Reply:
Now it is cheap again so not a big deal.